encode.go

```diff
@@ -211,13 +211,50 @@ func (e *encoder) fieldByIndex(v reflect.Value, index []int) (field reflect.Valu
 	return v
 }
 
-func (e *encoder) structv(tag string, in reflect.Value) {
+func (e *encoder) structv(tag string, in reflect.Value, head, line, foot []byte) {
 	sinfo, err := getStructInfo(in.Type())
 	if err != nil {
 		panic(err)
 	}
-	e.mappingv(tag, func() {
+
+	fieldsIndex := sinfo.FieldsList
+
+	headComments, lineComments, footComments := makeEmptyComments(len(fieldsIndex))
+
+	if fIndex := getYamlMeta(in, fieldsIndex); fIndex.IsValid() {
+		meta := fIndex.Elem().Interface().(StructMeta)
+		fieldsIndex = meta.GetFieldsIndex()
+		headComments, lineComments, footComments = meta.GetComments()
+	}
+
+	e.mappingv(tag, head, line, foot, func() {
+		processed := map[int]bool{}
+		for i, info := range fieldsIndex {
+			var value reflect.Value
+			if info.Inline == nil {
+				value = in.Field(info.Num)
+			} else {
+				value = e.fieldByIndex(in, info.Inline)
+				if !value.IsValid() {
+					continue
+				}
+			}
+			if info.OmitEmpty && isZero(value) {
+				continue
+			}
+			e.marshal("", reflect.ValueOf(info.Key), headComments[i][0], lineComments[i][0], footComments[i][0])
+			e.flow = info.Flow
+			e.marshal("", value, headComments[i][1], lineComments[i][1], footComments[i][1])
+			processed[info.Id] = true
+		}
+
 		for _, info := range sinfo.FieldsList {
+			if _, done := processed[info.Id]; done {
+				continue
+			}
+			if info.Key == "yaml_meta" {
```

Comment:
can we use a constant to declare `"yaml_meta"`? So that we can cross-reference through the constant

Reply:
👍